scala/lambdas/ingest-validate-generic-ingest-inputs/src/test/scala/uk/gov/nationalarchives/ingestvalidategenericingestinputs/MetadataJsonValueValidatorTest.scala

```diff
@@ -259,38 +262,43 @@ class MetadataJsonValueValidatorTest extends AnyFlatSpec with MockitoSugar with
     entriesGroupedById should equal(generateListOfIdsAndEntries(allEntries).sortBy(_._1))
   }
 
-  "getIdsOfAllEntries" should "still assign a file to an 'MetadataType' if it doesn't end in '-metadata.json' but is contained " +
-    "within the Asset's 'originalMetadataFiles' array" in {
-      val entriesWithMetadataFileWithNoExt = allEntries.map { entry =>
-        if entry(id).str == "d4f8613d-2d2a-420d-a729-700c841244f3" then Obj.from(entry.value ++ Map(name -> Str("TDD-2023-ABC-metadata"))) else entry
-      }
-      val entriesGroupedByType = entriesWithMetadataFileWithNoExt.groupBy(_(entryType).str)
-      val entriesAsValidatedMap = entriesGroupedByType.map { case (entryType, entries) =>
-        entryType -> entries.map(convertUjsonObjToSchemaValidatedMap)
-      }
+  forAll(casesWhereEntryShouldBeLabelledAsMetadataFile) { (metadataFileValueIsTrue, isInOriginalMetadataFiles) =>
+    "getIdsOfAllEntries" should s"still assign a file to a 'MetadataFileEntry' type if the 'metadataFile' property is '$metadataFileValueIsTrue' but its id $isInOriginalMetadataFiles contained " +
```

Comment:
The test name is quite hard to parse as it's so long as well so if you can shorten it a bit. If you try and keep it on one line that makes is easier to read.